internal/registry/registry.go

```diff
@@ -1106,6 +1110,27 @@ func (r *Registry) CheckSequence(peerID peer.ID, seq uint64) error {
 	return r.sequences.check(peerID, seq)
 }
 
+// periodicProviderReload reloads the registry's providers from the datastore.
+func (r *Registry) periodicProviderReload(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	for {
+		select {
+		case <-r.closing:
+			return
+		case <-ticker.C:
+			r.provMutex.Lock()
+
+			var err error
+			r.providers, err = loadPersistedProviders(context.Background(), r.dstore, r.filterIPs)
```

Comment:
I don't know how long this takes, but would it be better to load the new data and then lock and swap it out?

Reply:
but then you'd be overwriting any changes other writers could have done in the time between the datastore is read and the map is swapped. I think making other writers wait is worth it to avoid losing operations.

Reply:
What else is writing to this and why?

Reply:
I don't want find requests to be paused every 10 minutes just because we want to refresh the info. If there's writes because the ingest service is writing to this then it should be using a _different_ instance that is configured to not poll. The find service should not be doing any writes and should be configured to poll, without pausing reads for unnecessary periods.

Reply:
that's how it is configured. Ingesting is disabled in find tasks, which are configured to reload. At the same time, ingest tasks are configured to not reload.

No other requests _should_ write to the map, but the code base is complex and it's not written by us so I wanted to err on the side of caution. Besides, we don't know if this will ever become an actual performance issue.

But ok, I see your point, I'll update it.